rhea/system/memmap/barebone.py

```diff
@@ -11,61 +11,66 @@
 
 
 class Barebone(MemoryMapped):
+
+    r"""Generic memory-mapped interface.
+
+    This interface is the "generic" and most basic memory-map
+    interface.  This can be used for a common bus but also limited
+    functionality bus.  A bus cycle is instigated with a `write`
+    or `read` strobe.  The `write` and `read` strobes should
+    only be asserted if the `done` signal is active (high). As
+    soon as the peripheral (slave) detects the read or write
+    strobe the done signals will go low and remain low until
+    the slave has completed the transaction.
+
+    The done signals indicates two states, that the slave has
+    finished any previous operations and that it is ready to
+    retrieve any new operations.  A master can pulse the write
+    or read at anytime the done signals is active, the slave
+    will process one or more writes (up to max_burst).  At that
+    point the master needs to relinquish and wait for done.
+
+    The address bus is separated into two separate address buses,
+    peripheral address (`per_addr`, select a component/peripheral)
+    and the memory address (`mem_addr`).  The `per_addr` is used
+    to select a peripheral and the `mem_addr` is used to access
+    the memory-space in the peripheral.
+
+    The Barebone bus is point-to-point, only a single master
+    and slave can be connected to a bus.  The interconnect is
+    used to enable multiple slaves and masters
+
+    Timing Diagram
+    ---------------
+    | clock       /--\\__/--\\__/--\\__/--\\__/--\\__/--\\__/--\\__/--\\__
```

Comment:
As discussed on gitter, probably need to define an ascii-diagram standard to use in docstrings, preferably one that doesn't use backslahes.